debug.rst

```diff
@@ -1,8 +1,136 @@
+
+The ``app_dev.php`` front controller reads as follows by default::
+
+    // ...
+
+    $loader = require_once __DIR__.'/../app/bootstrap.php.cache';
+    require_once __DIR__.'/../app/AppKernel.php';
+
+    $kernel = new AppKernel('dev', true);
+    $kernel->loadClassCache();
+    $request = Request::createFromGlobals();
+
+To make your debugger happier, disable all PHP class caches by removing the
+call to ``loadClassCache()`` and by replacing the require statements like
+below::
+
+    // ...
+
+    // $loader = require_once __DIR__.'/../app/bootstrap.php.cache';
+    $loader = require_once __DIR__.'/../app/autoload.php';
+    require_once __DIR__.'/../app/AppKernel.php';
+
+    $kernel = new AppKernel('dev', true);
+    // $kernel->loadClassCache();
+    $request = Request::createFromGlobals();
+
+.. tip::
+
+    If you disable the PHP caches, don't forget to revert after your debugging
+    session.
+
+Some IDEs do not like the fact that some classes are stored in different
+locations. To avoid problems, you can either tell your IDE to ignore the PHP
+cache files, or you can change the extension used by Symfony for these files::
+
+    $kernel->loadClassCache('classes', '.php.cache');
+
+Dumping Variables with the VarDumper
+------------------------------------
+
+To ease the debugging of a variable in your application, you can use the
+:doc:`VarDumper component </components/var_dumper>` to dump the content of a
+variable. The component provides an alternative to the PHP :phpfunction:`var_dump()`
+function, in the form of ``dump()``::
+
+    // create a variable with a value
+    $myVar = ...;
+
+    // and dump it
+    dump($myVar);
+
+The dumper is not limited to scalar values. Arrays and objects can also be
+visualized using the VarDumper. One of the most important advantages of using
+``dump()`` is a nicer and more specialized dump of objects (e.g. Doctrine
+internals are filtered out when dumping a proxy entity).
+
+If the dumper is used on a command line, the result is a formatted string.
+Otherwise, the result is a piece of HTML, which can be expanded to show nested
+structures in the dumped value.
+
+You can also dump values from inside templates:
+
+.. code-block:: html+twig
+
+    {# dumps the variable inline as HTML #}
+    {{ dump(myVar) }}
+
+    {# dumps the variable to the web debug toolbar to not modify the template #}
+    {% dump myVar %}
+
+Useful Debugging Commands
+-------------------------
+
+When developing a large application, it can be hard to keep track of all the
+different services, routes and translations. Luckily, Symfony has some commands
+that can help you visualize and find the information.
+
+``debug:container``
+    Displays information about the contents of the Symfony container for all public
+    services. To find only those matching a name, append the name as an argument.
+
+``debug:config``
+    Shows all configured bundles, their class and their alias.
+
+``debug:router``
```

Comment:
Maybe we should reference here the `router:match` command too. In my experience it's very useful for newcomers and is very useful to quickly spot routing problems.